include/devicetree/fixed-partitions.h

```diff
@@ -63,6 +63,9 @@ extern "C" {
 
 /**
  * @brief Get a numeric identifier for a fixed partition
+ *
+ * The partition ID will be only available from status "okay" partition entries
```

Comment:
This is a backwards-incompatible change, and devicetree.h was just marked stable, so this will need to go through the stable API change process now, with an RFC issue, email to devel, etc. etc.

https://docs.zephyrproject.org/latest/reference/api/api_lifecycle.html#introducing-incompatible-changes